src/main/java/com/extrahardmode/ExtraHardMode.java

```diff
@@ -98,16 +92,37 @@ public void onEnable()
         registerModule(AntiFarming.class, new AntiFarming(this));
         registerModule(AnimalCrowdControl.class, new AnimalCrowdControl(this));
         registerModule(AntiGrinder.class, new AntiGrinder(this));
-        registerModule(DebugMode.class, new DebugMode(this));
+
+        //Disable debug if debug is disabled in all worlds. 
+        boolean ddbug = true;
+        for (World world : getServer().getWorlds())
+            if (!getModuleForClass(RootConfig.class).getBoolean(RootNode.DISABLE_DEBUG, world.getName()))
+            	ddbug = false;
+        if (ddbug){
+	        registerModule(DebugMode.class, new DebugMode(this));
+		}
+
         registerModule(Explosions.class, new Explosions(this));
         registerModule(HardenedStone.class, new HardenedStone(this));
         registerModule(LimitedBuilding.class, new LimitedBuilding(this));
         registerModule(MoreTnt.class, new MoreTnt(this));
         registerModule(Physics.class, new Physics(this));
         registerModule(Players.class, new Players(this));
-        registerModule(Torches.class, new Torches(this));
-        registerModule(Water.class, new Water(this));
+        
+        //Only register Torch module if at least one world has it active.
+        boolean dtorch = false;
+        for (World world : getServer().getWorlds()){
+        	final int torchMinY = getModuleForClass(RootConfig.class).getInt(RootNode.STANDARD_TORCH_MIN_Y, world.getName());
+        	if (torchMinY > 0){
```

Comment:
Pretty sure most of this project uses newline braces, fyi. Also, this entire block isn't indented for some reason.